src/services/iasql.ts

```diff
@@ -85,21 +87,29 @@ export async function connect(dbAlias: string, uid: string, email: string, dbId
 }
 
 export async function disconnect(dbAlias: string, uid: string) {
-  let conn, conn2;
+  let conn;
   try {
     const db: IasqlDatabase = await MetadataRepo.getDb(uid, dbAlias);
     conn = await createConnection(dbMan.baseConnConfig);
-    conn2 = await createConnection({
-      ...dbMan.baseConnConfig,
-      name: db.pgName,
-      database: db.pgName,
-    });
-    try {
-      await conn2.query(`SELECT * FROM query_cron('unschedule');`);
-      await conn2.query(`SELECT * FROM query_cron('unschedule_purge');`);
-    } catch (e) {
-      /** Do nothing */
-    }
+    // Cancel all connections
+    await conn.query(`REVOKE CONNECT ON DATABASE ${db.pgName} FROM PUBLIC, ${config.db.user}, ${db.pgUser};`);
```

Comment:
This line is doing a `REVOKE CONNECT` call on `config.db.user`. Then the code (in `unscheduleJobs`) is trying to connect to the same database using `dbMan.baseConnConfig` which also has its `username = config.db.user`. Am I missing something here or it should always fail?

Reply:
Good catch! Yes, I've messed up the order. We should try to unscheduled first, then revoke and cancel all ongoing connections.